### ISSUE TITLE
[HOLD for payment 2023-10-12] [$500] Web - User still can remove invalid member

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#27773]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Create a new workspace or open an existing one
2. Open the workspace & click on "Members"
3. Click on 'Invite' button
4. Search for an invalid phone number (+2523234211)
5. Select the user > click on "Next" > click on "Invite"
6. Select the user & click on the "Remove" button

## Expected Result:
Disable the checkbox for invalid entry as discuss here: https://github.com/Expensify/App/issues/24666#issuecomment-1701343822

## Actual Result:
User still can remove invalid member

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.71.6
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/94b70403-971d-4e24-a6e0-fcf49a773304

https://github.com/Expensify/App/assets/93399543/1025c00b-0e3b-435f-a4c5-06f78b1fa36d

**Expensify/Expensify Issue URL:**
**Issue reported by:** @DylanDylann
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1695099558494749

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c2f1b5f7d79b96d2</li>
        <li>Upwork Job ID: 1704104914090893312</li>
        <li>Last Price Increase: 2023-09-19</li>
<li>Automatic offers: </li>
<ul>
<li>cubuspl42 | Reviewer | 26897979</li>
<li>DylanDylann | Contributor | 26897980</li>
<li>DylanDylann | Reporter | 26897981</li>
</ul></li>
    </ul>
</details>

**Comment by kbecciv** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Still can remove invalid member
### What is the root cause of that problem?
We don’t disable checkbox if this member is invalid
### What changes do you think we should make in order to solve the problem?
We should disable checkbox of faulty users and even checkbox all if there are no other members
https://github.com/Expensify/App/blob/cb73cde0f0fe602b19eb924a4f8b11f759b079e9/src/pages/workspace/WorkspaceMembersPage.js#L311
In 2 places, we should add `|| !_.isEmpty(policyMember.errors)` to disable checkbox button if there is an error
### Result
https://github.com/Expensify/App/assets/141406735/f14594bb-6649-4339-98b8-b9031107f3ac

**Comment by adonis0302** [comment]
I wan to discuss this detail with you through slack.

**Comment by DylanDylann** [comment]
@cubuspl42 @sophiepintoraetz This issue is external but there are no C+ contributors assigned

**Comment by cubuspl42** [comment]
> Triggered auto assignment to Contributor-plus team member for initial proposal review - @cubuspl42 (`External`)

@DylanDylann It looks like it! This issue is in my queue, I'll review the proposal soon.

**Comment by cubuspl42** [comment]
Indeed, the typical interactions of the element with reduced opacity and an error are likely expected to be disabled, like in the case of a message with an error. The proposal by @DylanDylann makes sense.

C+ reviewed 🎀 👀 🎀

**Comment by sophiepintoraetz** [comment]
Waiting on confirmation from @jasperhuangg

**Comment by DylanDylann** [comment]
@jasperhuangg How I can move forward on this issue?

**Comment by jasperhuangg** [comment]
Sorry for the delay here, @DylanDylann's proposal makes sense to me, let's move forward!

**Comment by DylanDylann** [comment]
@cubuspl42 PR https://github.com/Expensify/App/pull/27991 is ready for review

**Comment by DylanDylann** [comment]
We requested a review from the internal engineer on 2023-09-27 but It seems jasperhuangg is busy on some days. It caused the timeline to last more than 3 days. So I think we are eligible for timeline bonus

**Comment by cubuspl42** [comment]
@DylanDylann 

> So I think we are eligible for timeline bonus

It's always up to the Expensify team to assign a bonus on a case-by-case basis, and formally, the PR should be put on hold while the clock is still running. It's better to just sum up your case and ask if the delay could be classified as a hold, than state that we are eligible 🙂

**Comment by DylanDylann** [comment]
@cubuspl42 Thanks so much for your reminder. 

>  It's better to just sum up your case and ask if the delay could be classified as a hold

This is exactly what I mean. Updated

**Comment by sophiepintoraetz** [comment]
FWIW, I agree - the delay came on our end so I think this qualifies for an urgency bonus!

**Comment by sophiepintoraetz** [comment]
Hey @anmurali - I'm headed OOO for the week so all that needs doing is payments on 12 Oct. This issue is eligible for the urgency bonus so the totals are:

@cubuspl42 requires payment [offer](https://www.upwork.com/nx/wm/offer/26897979) (Reviewer) - $750
@DylanDylann requires payment [offer](https://www.upwork.com/nx/wm/offer/26897980) (Contributor) - $800 ($50 for the reporting bonus and $750 for the contributor work)

**Comment by cubuspl42** [comment]
- The PR that introduced the bug has been identified. Link to the PR:
    -  It seems that the bug was present in the relevant components since they were created
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
    - N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
    - No need for additional discussion
- Determine if we should create a regression test for this bug.
    - Up to the QA team
- If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
  - Create or open an existing workspace.
  - Click on the "Members" section within the workspace.
  - Press the 'Invite' button.
  - Input an invalid phone number, such as "+2523234211" in the search bar.
  - Choose the found user, continue by clicking on "Next", and then click on "Invite".
  - Note that both the checkbox associated with +2523234211 and the 'Select All' checkbox are disabled.

**Comment by sophiepintoraetz** [comment]
all set here.
